src/main.controller.js

```diff
@@ -80,6 +97,63 @@ plugin.controller('wgnConfigCtrl', ['$scope', '$q', '$routeParams', 'znData', 'z
 			});
 		};
 
+
+		/**
+		 * Selects the configuration to copy.
+		 */
+		$scope.onSelectedConfig = function (config) {
+			if(previousSelectedConfig && Object.keys(previousSelectedConfig).length === 0){
+				config.selected = true;
+				selectedConfig = config;
+			} else {
+				if(previousSelectedConfig.name === config.name){
+					if(previousSelectedConfig.selected === true) {
+						config.selected = false;
+						selectedConfig = false;
+					} else {
+						config.selected = true;
+						selectedConfig = config;
+					}
+				} else {
+					config.selected = true;
+					selectedConfig = config;
+					if(previousSelectedConfig.selected === true) {
+						previousSelectedConfig.selected = false;
+					} 
+				}
+			}
+			previousSelectedConfig = config;
+		}
+
+		/**
+		 * Copy the current selected configuration.
+		 */
+		$scope.onCopyConfig = function () {
+			if (!selectedConfig && $scope.copyButtonText === 'Save Configuration') {
+				znMessage("Please select a configuration before saving!", "error");
+				$scope.copyButtonText = 'Copy Configuration';
+				$scope.copyConfigButtonSelected = false;
+				previousSelectedConfig.selected = false;
+			} else if (selectedConfig && $scope.copyButtonText === 'Save Configuration') {
+				$scope.copyButtonText = 'Copy Configuration';
+				$scope.copyConfigButtonSelected = false;
+				previousSelectedConfig.selected = false;
+
+				const copyConfiguration = Object.keys(selectedConfig).reduce(function (obj, key) {
+					if (key !== '$$hashKey' && key !== '$priority' && key !== '$id' && key !== 'selected' && !key.includes('webhookId') && !key.includes('webhookid') && !key.includes('WebhookId') && !key.includes('webhookKey') && !key.includes('webhookkey') && !key.includes('WebhookKey')) obj[key] = selectedConfig[key];
```

Comment:
please format this part. it's hard to folow. I would put a return charage after every && for better lisability